code/modules/vehicles/mecha/equipment/tools/work_tools.dm

```diff
@@ -303,18 +303,18 @@
 					return
 				qdel(target)
 		if(MODE_WALL)
-			if(isspaceturf(target))
-				var/turf/open/space/S = target
-				to_chat(source, "[icon2html(src, source)][span_notice("Building Floor...")]")
-				if(!do_after_cooldown(S, source))
-					return
-				S.PlaceOnTop(/turf/open/floor/plating, flags = CHANGETURF_INHERIT_AIR)
-			else if(isfloorturf(target))
+			if(isfloorturf(target))
 				var/turf/open/floor/F = target
 				to_chat(source, "[icon2html(src, source)][span_notice("Building Wall...")]")
 				if(!do_after_cooldown(F, source))
 					return
 				F.PlaceOnTop(/turf/closed/wall)
+			else if(isopenturf(target))
```

Comment:
Why change to isopenturf? This doesn't look better than isspaceturf considering we're casting to space turf.

Reply:
To cover all edge cases? I'm not sure I'll make it just  isspaceturf() then

Reply:
Ok it has to be isopenturf() becuase chasm isn't a subtype of space turf
https://github.com/tgstation/tgstation/blob/e16588b682a8a1516790d00ff85b305fcc62c11d/code/game/turfs/open/chasm.dm#L1-L3